sdk/src/test/java/com/mapbox/maps/renderer/FpsManagerTest.kt

```diff
@@ -0,0 +1,182 @@
+package com.mapbox.maps.renderer
+
+import android.os.Looper
+import android.view.Choreographer
+import com.mapbox.maps.logI
+import com.mapbox.maps.logW
+import com.mapbox.maps.renderer.FpsManager.Companion.IDLE_TIMEOUT_MS
+import io.mockk.*
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.Shadows
+import org.robolectric.annotation.LooperMode
+import org.robolectric.shadows.ShadowChoreographer
+import java.util.concurrent.TimeUnit
+
+@RunWith(RobolectricTestRunner::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class FpsManagerTest {
+
+  private lateinit var fpsManager: FpsManager
+
+  @Before
+  fun setUp() {
+    fpsManager = FpsManager()
+    mockkStatic("com.mapbox.maps.MapboxLogger")
+    every { logI(any(), any()) } just Runs
+    every { logW(any(), any()) } just Runs
+    // simulate all the tests on 60 Hz screen, meaning we have 16.6 ms time to render a frame
+    ShadowChoreographer.setPostFrameCallbackDelay(CHOREOGRAPHER_POST_FRAME_CALLBACK_DELAY_MS)
+    fpsManager.setScreenRefreshRate(SCREEN_FPS)
+  }
+
+  @Test
+  fun userRefreshRateNotSetTest() {
+    pauseHandler()
+    val frameTotalNumber = 10
+    var frameRenderedCount = 0
+    for (i in 0 until frameTotalNumber) {
+      Choreographer.getInstance().postFrameCallback {
+        assert(fpsManager.preRender(it))
+        frameRenderedCount++
+      }
+    }
+    idleHandler(vsyncCount = frameTotalNumber)
```

Comment:
It seems that all the `postFrameCallback` will be executed on the first Choreographer tick, since you post it before idling handler and all the runnables are executed together on the first onFrame callback. 
Checked the test locally and can confirm that `dleHandler(vsyncCount = 1)` still passes.